tests/test_key.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+"""Test cases for "key.py". """
+
+import copy
+import unittest
+
+import securesystemslib.formats
+import securesystemslib.keys as KEYS
+from securesystemslib.exceptions import (
+    CryptoError,
+    FormatError,
+    UnsupportedAlgorithmError,
+)
+from securesystemslib.key import SSlibKey
+from securesystemslib.signer import SSlibSigner
+
+
+class TestSSlibKey(unittest.TestCase):
+    """SSlibKey Test Case."""
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rsakey_dict = KEYS.generate_rsa_key()
+        cls.ed25519key_dict = KEYS.generate_ed25519_key()
+        cls.ecdsakey_dict = KEYS.generate_ecdsa_key()
+        cls.DATA_STR = "SOME DATA REQUIRING AUTHENTICITY."
+        cls.DATA = securesystemslib.formats.encode_canonical(cls.DATA_STR).encode(
+            "utf-8"
+        )
+
+    def test_sslib_verify(self):
+        """Test to check verify method of key."""
+
+        dicts = [self.rsakey_dict, self.ecdsakey_dict, self.ed25519key_dict]
+        for scheme_dict in dicts:
+            sslib_signer = SSlibSigner(scheme_dict)
+            sig_obj = sslib_signer.sign(self.DATA)
+
+            # Verify signature
+            sslib_key = SSlibKey.from_securesystemslib_key(scheme_dict)
+            verified = sslib_key.verify(sig_obj, self.DATA)
+            self.assertTrue(verified, "Incorrect signature.")
+
+            # Test for invalid public key.
+            public = scheme_dict["keyval"]["public"]
+            scheme_dict["keyval"]["public"] = ""
+            sslib_key = SSlibKey.from_securesystemslib_key(scheme_dict)
+
+            with self.assertRaises(
+                (CryptoError, UnsupportedAlgorithmError, FormatError)
+            ):
+                sslib_key.verify(sig_obj, self.DATA)
+
+            scheme_dict["keyval"]["public"] = public
+
+    def test_sslib_serialization(self):
+        """Test to check serialization methods of key."""
+
+        dicts = [self.rsakey_dict, self.ecdsakey_dict, self.ed25519key_dict]
+        for scheme_dict in dicts:
+            # Format key
+            key_dict = KEYS.format_keyval_to_metadata(
+                scheme_dict["keytype"],
+                scheme_dict["scheme"],
+                scheme_dict["keyval"],
+            )
+            # key_dict contains keyid_hash_algorithms.
+            key_dict.pop("keyid_hash_algorithms")
+
+            keyid = scheme_dict["keyid"]
+
+            # Test for load and dump key_dict
+            sslib_key = SSlibKey.from_dict(copy.copy(key_dict), keyid)
```

Comment:
Strange that `copy.copy` is enough here as `keyval` which is part of `key_dict` is usually a dictionary of itself.
Can you check that the copy operation works as expected?

Reply:
No changes occurred to `keyval` after popping out from `key_dict`, thats why it worked. But in Envelope signature get cleared because of https://github.com/in-toto/securesystemslib/blob/d3f48889595c6252ad30ccbec2b4c7d87795e37d/securesystemslib/signer.py#L65-L66

Reply:
It seems that using `copy.copy()` will work here as well.
I thought that because `keyval` is a "complicated" type - dict it will break the things, but it seems that dictionaries are considered as simple types which can be easily copied with `copy.copy` as well.
I tried this and it worked:
```
> import copy
> d = {"keyval": {"public": "abc"}}
> k = copy.copy(d)
> k
{'keyval': {'public': 'abc'}}